src/widgets/study-path/ui/stage-card/stage-card.test.tsx

```diff
@@ -0,0 +1,113 @@
+import { render, screen } from '@testing-library/react';
+import { describe, expect, it } from 'vitest';
+
+import { StageCard } from './stage-card';
+import { MOCKED_IMAGE_PATH } from '@/shared/__tests__/constants';
+import { StageCardProps } from '@/widgets/study-path/types';
+
+const mockProps: StageCardProps = {
+  id: 1,
+  title: 'Test title',
+  intro: 'Test description for the card.',
+  modules: [
+    {
+      id: 1,
+      text: 'Test module 1',
+      links: [
+        {
+          id: 1,
+          text: '',
+          title: 'Test link 1',
+          link: 'https://test1.com',
+          external: false,
+        },
+      ],
+      marked: false,
+    },
+    {
+      id: 2,
+      text: 'Test module 2',
+      links: [
+        {
+          id: 1,
+          text: '',
+          title: 'Test link 2',
+          link: 'https://test2.com',
+          external: false,
+        },
+      ],
+      marked: false,
+    },
+  ],
+  image: {
+    src: MOCKED_IMAGE_PATH,
+    alt: 'TestTitle',
+    className: 'stage-image',
+  },
+};
+
+describe('StageCard component', () => {
+  beforeEach(() => {
+    render(<StageCard {...mockProps} />);
+  });
+
+  it('displays correct stage number', () => {
+    const stageStep = screen.getByTestId('stage-step');
+
+    expect(stageStep).toBeVisible();
+    expect(stageStep.innerHTML).toBe(mockProps.id.toString());
```

Comment:
You could you `expect(stageStep).toHaveTextContent(mockProps.id.toString());`

Reply:
Done